File: spoj/3 - PRIME1/prime.cpp
#include <iostream>
#include <cmath>

using namespace std;

typedef long long ll;

bool isPrime(ll x) {
    if (x < 2)
        return false;
    if (x < 4)
        return true;
    if (x % 2 == 0)
        return false;
    if (x % 3 == 0)
        return false;
    for (ll i = 5; i < sqrt(x) + 2; i += 6) {
        if (x % i == 0)
            return false;
        if (x % (i + 2) == 0)
            return false;
    }
    return true;
}

int main() {
    ll t; cin >> t;
    while (t --) {
        ll m, n; cin >> m >> n;
        for (ll i = m; i <= n; i ++) {
            if (isPrime(i)) {
                cout << i << endl;
            }
        }
        if (t) cout << endl;
    }
}

File: spoj/3 - PRIME1/prime.py
from math import sqrt
def isPrime(x):
    if x < 2:
        return False
    if x < 4:
        return True
    if x % 2 == 0:
        return False
    if x % 3 == 0:
        return False
    for i in range(5, int(sqrt(n) + 2), 6):
        if x % i == 0:
            return False
        if x % (i + 2) == 0:
            return False
    return True

t = int(input())
for _ in range(t):
    m, n = map(int, input().split())
    for i in range(m, n + 1):
        if isPrime(i):
            print(i)
    if _ != t - 1:
        print("")